File: docs/clients/js/corridorClient.js
class CorridorClient {
  constructor(baseUrl = "http://localhost:5000/v1") {
    this.baseUrl = baseUrl;
  }

  async getCorridorsByGlyph(glyph) {
    const resp = await fetch(`${this.baseUrl}/corridors/glyph/${glyph}`);
    return await resp.json();
  }

  async getCorridorsByRci(band) {
    const resp = await fetch(`${this.baseUrl}/corridors/rci/${band}`);
    return await resp.json();
  }

  async getCorridorsByRemix(parentScroll) {
    const resp = await fetch(`${this.baseUrl}/corridors/remix/${parentScroll}`);
    return await resp.json();
  }

  async getCorridorMetadata(corridorId) {
    const resp = await fetch(`${this.baseUrl}/corridors/${corridorId}`);
    return await resp.json();
  }
}

export default CorridorClient;

File: modules/clients/python/corridor_client.py
import requests

class CorridorClient:
    def __init__(self, base_url="http://localhost:5000/v1"):
        self.base_url = base_url

    def get_corridors_by_glyph(self, glyph):
        url = f"{self.base_url}/corridors/glyph/{glyph}"
        resp = requests.get(url)
        resp.raise_for_status()
        return resp.json()

    def get_corridors_by_rci(self, band):
        url = f"{self.base_url}/corridors/rci/{band}"
        resp = requests.get(url)
        resp.raise_for_status()
        return resp.json()

    def get_corridors_by_remix(self, parent_scroll):
        url = f"{self.base_url}/corridors/remix/{parent_scroll}"
        resp = requests.get(url)
        resp.raise_for_status()
        return resp.json()

    def get_corridor_metadata(self, corridor_id):
        url = f"{self.base_url}/corridors/{corridor_id}"
        resp = requests.get(url)
        resp.raise_for_status()
        return resp.json()
